diff --git a/opt/atlassian/crowd/apache-tomcat/bin/setenv.sh b/opt/atlassian/crowd/apache-tomcat/bin/setenv.sh
index 5cfceeb..d4612e3 100755
--- a/opt/atlassian/crowd/apache-tomcat/bin/setenv.sh
+++ b/opt/atlassian/crowd/apache-tomcat/bin/setenv.sh
@@ -1,7 +1,23 @@
-JAVA_OPTS="-Xms128m -Xmx512m -Dfile.encoding=UTF-8 $JAVA_OPTS"
+#!/bin/bash
 
+CROWD_HOME="/var/atlassian/application-data/crowd"
+CROWD_CATALINA="/opt/atlassian/crowd"
+
+JAVA_OPTS="-Dfile.encoding=UTF-8 ${JAVA_OPTS}"
+JAVA_OPTS="-Xms2048m -Xmx2048m -XX:ReservedCodeCacheSize=512m ${JAVA_OPTS}"
+JAVA_OPTS="-Datlassian.plugins.enable.wait=300 -XX:+UnlockExperimentalVMOptions ${JAVA_OPTS}"
 export JAVA_OPTS
 
+CATALINA_OPTS="-Dcrowd.home=$CROWD_HOME ${CATALINA_OPTS}"
+CATALINA_OPTS="-XX:+UseZGC -XX:+ExplicitGCInvokesConcurrent ${CATALINA_OPTS}"
+CATALINA_OPTS="-Dcrowd.catalina.connector.port=8095 ${CATALINA_OPTS}"
+CATALINA_OPTS="-Dcrowd.catalina.connector.scheme=http ${CATALINA_OPTS}"
+CATALINA_OPTS="-Dcrowd.catalina.connector.secure=false ${CATALINA_OPTS}"
+CATALINA_OPTS="-Dcrowd.catalina.connector.proxyname= ${CATALINA_OPTS}"
+CATALINA_OPTS="-Dcrowd.catalina.connector.proxyport= ${CATALINA_OPTS}"
+CATALINA_OPTS="-Dcrowd.catalina.context.path= ${CATALINA_OPTS}"
+export CATALINA_OPTS
+
 # set the location of the pid file
 if [ -z "$CATALINA_PID" ] ; then
     if [ -n "$CATALINA_BASE" ] ; then
diff --git a/opt/atlassian/crowd/apache-tomcat/conf/server.xml b/opt/atlassian/crowd/apache-tomcat/conf/server.xml
index 1ba6efb..f011824 100644
--- a/opt/atlassian/crowd/apache-tomcat/conf/server.xml
+++ b/opt/atlassian/crowd/apache-tomcat/conf/server.xml
@@ -10,12 +10,15 @@
                    maxHttpHeaderSize="8192"
                    maxThreads="150"
                    minSpareThreads="25"
-                   port="8095"
+                   port="${crowd.catalina.connector.port}"
                    redirectPort="8443"
                    useBodyEncodingForURI="true"
                    URIEncoding="UTF-8"
                    compression="on"
                    sendReasonPhrase="true"
+                   relaxedPathChars="[]|" relaxedQueryChars="[]|{}^&#x5c;&#x60;&quot;&lt;&gt;"
+                   scheme="${crowd.catalina.connector.scheme}" secure="${crowd.catalina.connector.secure}"
+                   proxyName="${crowd.catalina.connector.proxyname}" proxyPort="${crowd.catalina.connector.proxyport}"
                    compressableMimeType="text/html,text/xml,application/xml,text/plain,text/css,application/json,application/javascript,application/x-javascript" />
 
         <Engine defaultHost="localhost" name="Catalina">
@@ -23,6 +25,9 @@
                 <Valve className="org.apache.catalina.valves.ErrorReportValve"
                        showReport="true"
                        showServerInfo="false" />
+                <Context path="${crowd.catalina.context.path}" docBase="../../crowd-webapp">
+                    <Manager pathname="" />
+                </Context>
             </Host>
         </Engine>
 
diff --git a/opt/atlassian/crowd/start_crowd.sh b/opt/atlassian/crowd/start_crowd.sh
index f4f3c60..961d0aa 100755
--- a/opt/atlassian/crowd/start_crowd.sh
+++ b/opt/atlassian/crowd/start_crowd.sh
@@ -1,6 +1,19 @@
-#!/bin/sh
-set -e
+#!/bin/bash
 
+# resolve links - $0 may be a softlink
+PRG="$0"
+while [ -h "$PRG" ]; do
+    ls=`ls -ld "$PRG"`
+    link=`expr "$ls" : '.*-> \(.*\)$'`
+    if expr "$link" : '/.*' > /dev/null; then
+        PRG="$link"
+    else
+        PRG=`dirname "$PRG"`/"$link"
+    fi
+done
+PRGDIR=`dirname "$PRG"`
+
+PRGRUNMODE=false
 if [ "$1" = "-fg" ] || [ "$1" = "run" ]  ; then
 	shift
 	PRGRUNMODE=true
@@ -8,10 +21,8 @@ else
 	echo "To run Crowd in the foreground, start the server with $0 -fg"
 fi
 
-cd "`dirname "$0"`"
-
-if [ "$PRGRUNMODE" = "true" ] ; then
-    exec apache-tomcat/bin/catalina.sh run "$@"
+if [ "$PRGRUNMODE" == "true" ] ; then
+    exec $PRGDIR/apache-tomcat/bin/catalina.sh run "$@"
 else
-    exec apache-tomcat/bin/startup.sh "$@"
-fi
\ No newline at end of file
+    exec $PRGDIR/apache-tomcat/bin/startup.sh "$@"
+fi
